### PR TITLE
docs: clarify exec endpoint usage in http

### PR DESCRIPTION
**Pull request by JoanFM**
**Goals**

Clarify confusion between HTTP endpoints and Executor endpoints

### CODE REVIEW

File: docs/fundamentals/gateway/index.md [hunk 1]
```diff
@@ -205,6 +205,24 @@ with:
...
+However, if you want to send requests to a different Executor endpoint, you can still do it without exposing it in the HTTP endpoint, by sending an HTTP request to the `/post` HTTP endpoint while setting  
+`execEndpoint` in the request.
+
+```text
+curl --request POST \
+'http://localhost:12345/post' \
+--header 'Content-Type: application/json' -d '{"data": [{"text": "hello world"}], "execEndpoint": "/foo"}'
+```
+
+this would be equivalent to passing the `on` parameter to `client.post` like:
```
Comment (alexcg1) [inline docs/fundamentals/gateway/index.md:218]:
```suggestion
The above cURL command is equivalent to passing the `on` parameter to `client.post` as follows:
```

File: docs/fundamentals/gateway/index.md [hunk 2]
```diff
@@ -205,6 +205,24 @@ with:
...
+
+```python
+from jina import Client, DocumentArray, Document
```
Comment (alexcg1) [inline docs/fundamentals/gateway/index.md:221]: Should we be importing everything from `jina` or some from `docarray?`

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Looking good. Just a small change and question

File: docs/fundamentals/gateway/index.md (hunk 2 above)
Comment (JoanFM) [reply docs/fundamentals/gateway/index.md:221]: I like `docarray` good point

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

**Pull request merged**